### DISCUSSION TITLE
How to include a softwareSystemInstance into a deployment view?

### DISCUSSION THREAD

**Discussion by AndreiPashkin-SoftTeco** [structurizr/java#234] [Q&A]
I have slightly modified an example of a deployment view and included a `softwareSystemInstance` in it. And it makes the diagram look like that:

![image](https://github.com/structurizr/java/assets/153719529/d5229c19-03aa-411a-846c-c7ecd0fa1f32)

Both `webapp` and `database` containers are not visible on it anymore. And the only way to make them visible that I've found is to remove `softwareSystemInstance s` statement from it.

I wonder if there is a way to show both `softwareSystemInstance` and `containerIsntance` elements contained within it?

Here is the code:
```
workspace {

    model {
        u = person "User"
        s = softwareSystem "Software System" {
            webapp = container "Web Application" "" "Spring Boot"
            database = container "Database" "" "Relational database schema"
        }

        u -> webapp "Uses"
        webapp -> database "Reads from and writes to"
        
        development = deploymentEnvironment "Development" {
            deploymentNode "Cluster" {
                softwareSystemInstance s
                
                deploymentNode "Developer Laptop" {
                    containerInstance webapp
                    deploymentNode "MySQL" {
                        containerInstance database
                    }
                }
            }
        }
    }

    views {
        deployment * development {
            include *
            autoLayout lr
        }
    }
    
}
```

**Comment by simonbrowndotje** [comment]
> I wonder if there is a way to show both softwareSystemInstance and containerIsntance elements contained within it?

You can only show one of those levels of abstraction on the same diagram I'm afraid.
